feat: shortcut deprecation section (#4537)
JetBrains/kotlin-web-site: docs/topics/multiplatform/multiplatform-compatibility-guide.md

kotlin {
    jvm()
    js()

>     iosArm64()

Reviewer: Let's also add `iosSimulatorArm64()`

kotlin {
    jvm()
    js()

>     iosArm64()

Reviewer: and here

  with "unresolved reference" errors, and binaries (for example, Gradle plugins) might fail with linkage errors
  unless recompiled against the latest versions of the Kotlin Gradle plugin

<anchor name="target-shortcuts-deprecation"/>
## Deprecated Apple target shortcuts

**What's changed?**

We're deprecating `ios()`, `watchos()`, and `tvos()` target shortcuts in Kotlin Multiplatform DSL. They were designed to
partially create a source set hierarchy for Apple targets. However, they proved to be difficult to expand and sometimes confusing.

For example, the `ios()` shortcut created both the `iosArm64` and `iosX64` targets but didn't include the `iosSimulatorArm64`
target, which is necessary when working on hosts with Apple M chips. However, changing this shortcut was hard to implement
and could cause issues in existing user projects.

**What's the best practice now?**

The Kotlin Gradle plugin now provides a built-in hierarchy template. Since Kotlin 1.9.20, it's enabled by default
and contains predefined intermediate source sets for popular use cases.

> The difference from shortcuts is that you need to explicitly specify the set of targets first, and the plugin then

Reviewer 2: ```suggestion
Instead of shortcuts you should specify the list of targets, and then the plugin
```

"Explicitly specify" reads redundant to me here, but maybe it isn't?
Also, changed "set" to "list" so that it corresponds to "this list" later.

source sets based on this list.

For example, if you have `iosArm64` and `iosSimulatorArm64` targets in your project, the plugin automatically creates

> the `iosMain` and `iosTest` intermediate source sets. For the `iosArm64` and `macosArm64` targets, the `appleMain` and

Reviewer 2: ```suggestion
the `iosMain` and `iosTest` intermediate source sets. If you have `iosArm64` and `macosArm64` targets, the `appleMain` and
```

I would suggest mirroring the sentence structure here so that it's clearer that this is not a continuation but a separate hypothetical scenario.

> targets, it creates the `appleMain` and `appleTest` source sets.

Reviewer 2: I don't think I understand where the `macosArm64` target comes from, if we didn't suppose that the project has it from the beginning. Is "`iosArm64` and `iosSimulatorArm64`" one example and "`iosArm64` and `macosArm64`" another?

Author: Yes, another example to show that different intermediate source sets are created, I should probably split it into two senetences
